### PR TITLE
Adding prometheus-to-sd chart

### PR DESCRIPTION
**Pull request by acondrat**
The chart deploys [prometheus-to-sd](https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd) with an optional list of metrics sources to scrape from.

### CODE REVIEW

File: stable/prometheus-to-sd/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+description: A Helm chart for Kubernetes
```
Comment (fhemberger) [inline stable/prometheus-to-sd/Chart.yaml]: This should be the actual description of the chart. ;)

Comment (acondrat) [reply stable/prometheus-to-sd/Chart.yaml]: Thanks for reviewing. Description updated.

File: stable/prometheus-to-sd/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+| Parameter                             | Description                           | Default                                                   |
+| --------------------------            | ------------------------------------- | --------------------------------------------------------- |
+| `image.repository`                    | prometheus-to-sd image repository     | `gcr.io/google-containers/prometheus-to-sd`               |
+| `image.tag`                           | prometheus-to-sd image tag            | `v0.2.1`                                                  |
+| `image.pullPolicy`                    | Image pull policy                     | `IfNotPresent`                                            |
+| `resources`                           | CPU/Memory resource requests/limits   | Memory: `64Mi`, CPU: `50m`                                |
+| `port`                                | Profiler port                         | 6060                                                      |
+| `metricSources`                       | Sources for metrics in the next format: component-name:http://host:port?whitelisted=a,b,c | {}    |
```
Comment (unguiculus) [inline stable/prometheus-to-sd/README.md]: Please align table nicely.

File: stable/prometheus-to-sd/values.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+replicaCount: 1
+image:
+  repository: gcr.io/google-containers/prometheus-to-sd
+  tag: v0.2.1
+  pullPolicy: IfNotPresent
+resources:
```
Comment (unguiculus) [inline stable/prometheus-to-sd/values.yaml]: We usually recommend not to specify defaults for resources and leave this a conscious choice of the user.

File: stable/prometheus-to-sd/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+description: Scrape metrics stored in prometheus format and push them to the Stackdriver
+name: prometheus-to-sd
```
Comment (unguiculus) [inline stable/prometheus-to-sd/Chart.yaml]: Add `appVersion`, `maintainers`, `home`, `icon`, `source`, `keywords`, etc. if possible.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR. Note that we've started namespacing templates (see #1785). Please apply this to your PR.

**Comment by acondrat** [comment]
@unguiculus thanks for the review! Suggested changes applied.

File: stable/prometheus-to-sd/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}
```
Comment (unguiculus) [inline stable/prometheus-to-sd/templates/deployment.yaml]: Why did you change this? Should be `{{ template "prometheus-to-sd.fullname" . }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please fix conflicts. Something got messed up. You should not have any test script changes in the PR.

File: stable/prometheus-to-sd/templates/deployment.yaml (hunk 5 above)
Comment (acondrat) [reply stable/prometheus-to-sd/templates/deployment.yaml]: `{{ template "prometheus-to-sd.fullname" . }}` causes ugly resource names. The problem is discussed in https://github.com/kubernetes/helm/issues/2062  and I think it might be solved in Helm 3 if  https://github.com/kubernetes/helm/issues/2060 is approved. `{{ .Release.Name }}` is unique and can be safely used as the name of the resource.

Comment (acondrat) [reply stable/prometheus-to-sd/templates/deployment.yaml]: I had to cherry-pick some commits related to circle CI because the PR was opened long before test changes applied, so the tests were failing.

File: stable/prometheus-to-sd/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+apiVersion: extensions/v1beta1
```
Comment (unguiculus) [inline stable/prometheus-to-sd/templates/deployment.yaml]: apiVersion: apps/v1beta1

File: stable/prometheus-to-sd/templates/deployment.yaml (hunk 5 above)
Comment (unguiculus) [reply stable/prometheus-to-sd/templates/deployment.yaml]: Currently, the convention is to use `{{ template "prometheus-to-sd.fullname" . }}`. This is what `helm create` gives you. Chart examples use this as well. Please follow this best practice. You may adjust the implementation of the template providing a fullnameOverride if you like. Compare https://github.com/kubernetes/charts/blob/master/incubator/common/templates/_fullname.tpl#L20-L29.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please rebase.

File: stable/prometheus-to-sd/templates/deployment.yaml (hunk 5 above)
Comment (unguiculus) [reply stable/prometheus-to-sd/templates/deployment.yaml]: Here's another alternative that should work for you.
https://github.com/compleatang/charts/blob/7cfe269290af6c2cc1e0ae9e801ad31a9099ba8d/incubator/oauth-proxy/templates/_helpers.tpl#L13-L19

File: stable/prometheus-to-sd/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+appVersion: 0.2.2
+description: Scrape metrics stored in prometheus format and push them to the Stackdriver
+home: https://github.com/GoogleCloudPlatform/k8s-stackdriver/tree/master/prometheus-to-sd
+keywords:
+- prometheus
+- stackdriver
+maintainers:
+- name: Arcadie Condrat
```
Comment (unguiculus) [inline stable/prometheus-to-sd/Chart.yaml]: Use Github username.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/prometheus-to-sd/values.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+replicaCount: 1
+image:
+  repository: gcr.io/google-containers/prometheus-to-sd
+  tag: v0.2.2
+  pullPolicy: IfNotPresent
+resources: {}
```
Comment (unguiculus) [inline stable/prometheus-to-sd/values.yaml:6]: Add empty default for nodeSelector and add it to the readme.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by acondrat** [comment]
The reason for e2e tests failure is probably `Failed to get GCE config: Not running on GCE.`. I don't see the POD's log to say for sure.
prometheus-to-sd can only run in GCE.

**Review by unguiculus — APPROVED** [review]
